core/src/main/java/org/apache/iceberg/ManifestFilterManager.java

```diff
@@ -86,7 +86,10 @@ public String partition() {
   private final Map<ManifestFile, Iterable<F>> filteredManifestToDeletedFiles =
       Maps.newConcurrentMap();
 
-  protected ManifestFilterManager(Map<Integer, PartitionSpec> specsById) {
+  // tracking the min sequence number of data files on each partition
+  private final Map<String, Long> minSequenceNumberByPartition = Maps.newHashMap();
```

Comment:
There I use the `String` as the key of the map because `StructLikeMap` only supports one partition spec. When we scan the data manifest files, there may be multi-different partition specs.